hadoop-shims/common/src/main/java/org/apache/uniffle/hadoop/shim/HadoopShimsLoader.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.hadoop.shim;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ServiceLoader;
+import java.util.StringTokenizer;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.util.VersionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HadoopShimsLoader {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HadoopShimsLoader.class);
```

Comment:
Actually, we don't need the class HadoopShimsLoader. We can refer to Spark's implement. We can use the class `HadoopShims` directly.

Reply:
> Maybe I am not catch you idea. I think hadoop's implement is not same with spark's implement. Spark packages are only used in module client-spark/spark*, but hadoop packages are used everywhere. So for spark, we know the real spark version whether in client-spark/spark2 or client-spark/spark3. But for hadoop, we can't assume we know the hadoop version. so we need HadoopShimsLoader to construct the right HadoopShim.

But we use one certain hadoop profile, we can add the certain class.